src/core_plugins/kibana/public/context/api/anchor.js

```diff
@@ -0,0 +1,27 @@
+import _ from 'lodash';
+
+import { addComputedFields } from './utils/fields';
+import { createAnchorQuery } from './utils/queries';
+
+
+async function fetchAnchor(es, indexPattern, uid, sort) {
+  const indices = await indexPattern.toIndexList();
+  const response = await es.search({
+    index: indices,
+    body: addComputedFields(indexPattern, createAnchorQuery(uid, sort)),
+  });
+
+  if (_.get(response, ['hits', 'total'], 0) < 1) {
```

Comment:
I would at least generally translate the ES errors into something more generic at the service level, so that clients of the service don't end up coupling themselves to the ES client.